File: shared/types/photo.ts
// Photo-related TypeScript type definitions

export interface Photo {
  id: string;
  filename: string;
  original_filename: string;
  file_path: string;
  thumbnail_path?: string;
  caption?: string;
  latitude: number;
  longitude: number;
  altitude?: number;
  created_at: string;
  updated_at: string;
  user_id?: string;
}

export interface PhotoUpload {
  file: File;
  caption?: string;
  latitude: number;
  longitude: number;
}

export interface PhotoFilters {
  latitude?: number;
  longitude?: number;
  radius?: number;
  page?: number;
  per_page?: number;
}

export interface PhotoStats {
  total_photos: number;
  recent_photos: number;
}

export interface PhotoResponse {
  photos: Photo[];
  total: number;
  page: number;
  per_page: number;
  pages: number;
}

File: server/app/models/photo.py
from app import db
from datetime import datetime
import uuid

class Photo(db.Model):
    __tablename__ = 'photos'
    
    id = db.Column(db.String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    filename = db.Column(db.String(255), nullable=False)
    original_filename = db.Column(db.String(255), nullable=False)
    file_path = db.Column(db.String(500), nullable=False)
    thumbnail_path = db.Column(db.String(500))
    caption = db.Column(db.Text)
    latitude = db.Column(db.Float, nullable=False)
    longitude = db.Column(db.Float, nullable=False)
    altitude = db.Column(db.Float)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    user_id = db.Column(db.String(36), db.ForeignKey('users.id'), nullable=True)
    
    # Relationships
    user = db.relationship('User', backref='photos')
    
    def to_dict(self):
        return {
            'id': self.id,
            'filename': self.filename,
            'original_filename': self.original_filename,
            'file_path': self.file_path,
            'thumbnail_path': self.thumbnail_path,
            'caption': self.caption,
            'latitude': self.latitude,
            'longitude': self.longitude,
            'altitude': self.altitude,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'updated_at': self.updated_at.isoformat() if self.updated_at else None,
            'user_id': self.user_id
        }
    
    def __repr__(self):
        return f'<Photo {self.id}: {self.original_filename}>'
